app/models/ability.rb

```diff
@@ -0,0 +1,45 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    return unless user.present?
+
+    can :read, :all
+
+    can %i[create destroy], Post, author: user
+
+    can %i[create destroy], Comment, author: user
```

Comment:
- [Required] Please consider making sure that a `user` who has the role of `admin` can delete any `comment` or `post`, I changed the role of a `user` to `admin` and I was not able to delete a `post`. I believe that if you refactor this line of code as shown below the `authorizations` will work as expected, please see the code snippet below

```ruby
class Ability
  include CanCan::Ability

  def initialize(user)
    return unless user.present?

    can :read, :all

    if user.role == 'admin' 
      can :manage, :all
    else
      can :manage, Post, author: user 
      can :manage, Comment, author: user
      can :read, :all
    end
  end
end
```
The code above will give authorization to an `admin` to `delete` all `posts` and `comments` available in the application and other users will only be able to delete their posts and comments.